src/components/molecules/pokemonCard/PokemonCard.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import Axios from 'axios';
+
+import { Card, Header, ImageContainer } from '../../atoms';
+import { Ability } from '../ability/Ability';
+
+import './PokemonCard.scss';
+
+export const PokemonCard = ({ url, name }) => {
+	const [avatar, setAvatar] = useState(null);
+	const [moves, setMoves] = useState(0);
+	const [weight, setWeight] = useState(0);
+	const [abilities, setAbilities] = useState(0);
+	const [isLoading, setIsLoading] = useState(false);
+
+	const cancelToken = Axios.CancelToken.source();
+
+	useEffect(() => {
+		const fetchCardData = async () => {
+			try {
+				setIsLoading(true);
+
+				const responseData = await Axios.get(`${url}`, {
+					cancelToken: cancelToken.token
+				}).catch((thrown) => {
+					if (Axios.isCancel(thrown)) {
+						console.log(thrown.message);
+					} else {
+						console.log('Handle error');
+					}
+				});
+
+				const {
+					data: {
+						abilities,
+						weight,
+						moves,
+						sprites: { front_default }
+					}
+				} = responseData;
+				setAbilities(abilities);
+				setWeight(weight);
+				setMoves(moves.length);
+				setAvatar(front_default);
+				setIsLoading(false);
+			} catch (error) {
+				console.log('ERROR___Pokemoncard__:', error);
+				setIsLoading(false);
+			}
+		};
+
+		fetchCardData();
+
+		return () => {
+			console.log('Cleanup Pokemoncard');
+			setIsLoading(false);
+		};
+	}, [url]);
+
+	return (
+		<>
+			{!isLoading && (
```

Comment:
Als we niet meer aan het laden zijn, zijn we niet zeker dat we pokemon data hebben, toch? We kunnen ook een error teruggekregen hebben en dan is onze pokemon data nog `null` of `0`